mm2src/coins/tendermint/tendermint_tx_history_v2.rs

```diff
@@ -0,0 +1,754 @@
+use super::iris::htlc_proto::QueryHtlcResponseProto;
+use super::{rpc::*, type_urls::*, AllBalancesResult, TendermintCoin, TendermintCoinRpcError, TendermintToken};
+
+use crate::my_tx_history_v2::{CoinWithTxHistoryV2, MyTxHistoryErrorV2, MyTxHistoryTarget, TxHistoryStorage};
+use crate::tendermint::iris::htlc::{HTLC_STATE_COMPLETED, HTLC_STATE_OPEN, HTLC_STATE_REFUNDED};
+use crate::tendermint::iris::htlc_proto::{ClaimHtlcProtoRep, CreateHtlcProtoRep};
+use crate::tendermint::TendermintFeeDetails;
+use crate::tx_history_storage::{GetTxHistoryFilters, WalletId};
+use crate::utxo::utxo_common::big_decimal_from_sat_unsigned;
+use crate::{HistorySyncState, MarketCoinOps, TransactionDetails, TxFeeDetails};
+use async_trait::async_trait;
+use common::executor::Timer;
+use common::log;
+use common::state_machine::prelude::*;
+use cosmrs::proto::cosmos::bank::v1beta1::MsgSend;
+use cosmrs::proto::cosmos::base::v1beta1::Coin as CoinAmount;
+use cosmrs::tx::Fee;
+use mm2_err_handle::prelude::{MmError, MmResult};
+use mm2_metrics::MetricsArc;
+use mm2_number::BigDecimal;
+use primitives::hash::H256;
+use prost::Message;
+use rpc::v1::types::Bytes as BytesJson;
+use std::collections::{hash_map::Entry, HashMap};
+
+macro_rules! try_or_return_stopped_as_err {
+    ($exp:expr, $reason: expr, $fmt:literal) => {
+        match $exp {
+            Ok(t) => t,
+            Err(e) => {
+                return Err(Stopped {
+                    phantom: Default::default(),
+                    stop_reason: $reason(format!("{}: {}", $fmt, e)),
+                })
+            },
+        }
+    };
+}
+
+#[async_trait]
+pub trait TendermintTxHistoryOps: CoinWithTxHistoryV2 + MarketCoinOps + Send + Sync + 'static {
+    async fn get_rpc_client(&self) -> MmResult<HttpClient, TendermintCoinRpcError>;
+
+    async fn query_htlc(&self, id: String) -> MmResult<QueryHtlcResponseProto, TendermintCoinRpcError>;
+
+    fn decimals(&self) -> u8;
+
+    fn set_history_sync_state(&self, new_state: HistorySyncState);
+
+    async fn all_balances(&self) -> MmResult<AllBalancesResult, TendermintCoinRpcError>;
+
+    fn all_wallet_ids(&self) -> Vec<WalletId>;
+
+    // Returns Hashmap where key is denom and value is ticker
+    fn get_denom_and_ticker_map(&self) -> HashMap<String, String>;
+}
+
+#[async_trait]
+impl CoinWithTxHistoryV2 for TendermintCoin {
+    fn history_wallet_id(&self) -> WalletId { WalletId::new(self.ticker().replace('-', "_")) }
```

Comment:
Yes, we do.

Reply:
Could you please clarify what's the reason for that? I could miss something

Reply:
Using `-` directly on sqlite table name isn't valid. If I remember correctly you can use them if you put put brackets or double quotes around the table name which is probably not implemented in our case since sql breaks on those table names.

Reply:
Oh I forgot that we use the WalletId as a part of table names... Could we move this logic from `CoinWithTxHistoryV2::history_wallet_id` to `WalletId::new`?
I believe it could fix this issue for all coins. It's strange that we still don't have any report. For example, for `USDT-SLP` coin.
@artemii235

Reply:
already did https://github.com/KomodoPlatform/atomicDEX-API/blob/8010f46ce9e24a9509e02ee39d74096011989963/mm2src/coins/tx_history_storage/mod.rs#L105-L112

Reply:
Nice, thank you!